iam.tf

```diff
@@ -10,22 +11,15 @@ resource "aws_iam_role" "this" {
       Principal = { Service = "ec2.amazonaws.com" }
     }]
   })
-}
 
-resource "aws_iam_role_policy_attachment" "this" {
-  for_each = toset([
+  managed_policy_arns = [
```

Comment:
I'm not sure about this change - the [aws_iam_role docs](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/iam_role) recommend using the separate resource.

Maybe we should just adjust the `for_each` to use an empty set if `create_iam_role` is `false`?

Reply:
Funny enough, that was the original implementation but I felt it was unnecessarily complex and simplified it but I hear you concerns and will revert to that implementation.

Reply:
Done